R/ggplotMDS.R

```diff
@@ -70,22 +73,62 @@ ggplotMDS <- function(DGEdata,
                       sizeBy,
                       top = Inf,
                       labels,
-                      labelSize,
+                      labelSize = 3,
                       title,
                       hlineIntercept,
                       vlineIntercept,
-                      reflineColor = "darkgoldenrod1",
+                      reflineColor = "red",
                       reflineSize = 0.5,
-                      symShape = 16,
-                      symSize = 5,
-                      alpha = 0.7,
-                      shapes,
-                      colors,
+                      symShape = "circle",
+                      symSize = 10,
+                      transparency = 0.7,
                       dim.plot = c(1, 2)) {
 
-    # Default labels to colnames of DGEdata
+    plotType = tolower(plotType)
+    assertthat::assert_that(class(DGEdata) %in% c("DGEobj", "DGEList", "matrix", "Dataframe"),
+                            msg = "DGEdata must be of class 'DGEList', 'DGEobj', 'matrix' or 'Dataframe'.")
+    assertthat::assert_that(plotType %in% c("canvasxpress", "ggplot"),
+                            msg = "Plot type must be either canvasXpress or ggplot.")
+    assertthat::assert_that(!missing(colorBy),
+                            length(colorBy) == ncol(DGEdata),
+                            msg = "colorBy must be specified and should be the length of the number of columns in DGEdata.")
+
+    if (!missing(shapeBy)) {
+        assertthat::assert_that(length(shapeBy) == ncol(DGEdata),
+                                msg = "shapeBy should be the length of the number of columns in DGEdata.")
+    }
+
+    if (!missing(sizeBy)) {
+        assertthat::assert_that(length(sizeBy) == ncol(DGEdata),
+                                msg = "sizeBy should be the length of the number of columns in DGEdata.")
+    }
+
+    if (any(length(top) != 1, !is.numeric(top))) {
+        warning("top should be a numeric value or Inf. Assigning default value 'Inf'.")
+        top <- Inf
+    }
+
+    if (any(!is.numeric(transparency), length(transparency) != 1, transparency < 0, transparency > 1)) {
+        warning("transparency must be a singular numeric value and must be between 0 and 1. Assigning default value 0.7.")
+        symbolTransparency <- 0.7
+    }
+
+    # Validate labels
     addLabels <- TRUE
-    if (missing(labels)) {
+    addDefaultLabel <- FALSE
+
+    if (!missing(labels)) {
+        if (is.null(labels)) {
+            addLabels <- FALSE
+        } else if (!length(labels) == ncol(DGEdata)) {
+            warning("Number of labels does not match the number of columns in DGEdata. Assigning default values.")
```

Comment:
The default values differ. In this case it could either be colnames of the input dataframe or replicate group if a DGEobj is passed. Hence I have left it blank.